src/mono/mono/metadata/marshal-lightweight.c

```diff
@@ -2319,6 +2320,57 @@ emit_array_accessor_wrapper_ilgen (MonoMethodBuilder *mb, MonoMethod *method, Mo
 	mono_mb_emit_byte (mb, CEE_RET);
 }
 
+static void
+emit_unsafe_accessor_field_wrapper (MonoMethodBuilder *mb, MonoMethod *accessor_method, MonoMethodSignature *sig, MonoGenericContext *ctx, MonoUnsafeAccessorKind kind, const char *member_name)
+{
+	// Field access requires a single argument for target type and a return type.
+	g_assert (kind == MONO_UNSAFE_ACCESSOR_FIELD || kind == MONO_UNSAFE_ACCESSOR_STATIC_FIELD);
+	g_assert (member_name != NULL);
+
+	MonoType *target_type = sig->params[0]; // params[0] is the accessor wrapper's parent
+	MonoType *ret_type = sig->ret;
+	if (sig->param_count != 1 || target_type == NULL || sig->ret->type == MONO_TYPE_VOID)
+		mono_mb_emit_exception_full (mb, "System", "BadImageFormatException", "UnsafeAccessor");
+
+	MonoClass *target_class = mono_class_from_mono_type_internal (target_type);
+	gboolean target_byref = m_type_is_byref (target_type);
+	gboolean target_valuetype = m_class_is_valuetype (target_class);
+	gboolean ret_byref = m_type_is_byref (ret_type);
+	if (!ret_byref || (kind == MONO_UNSAFE_ACCESSOR_FIELD && target_valuetype && !target_byref))
+		mono_mb_emit_exception_full (mb, "System", "BadImageFormatException", "UnsafeAccessor");
+
+	mono_mb_emit_ldarg (mb, 0);
+	
+	MonoClassField *target_field = mono_class_get_field_from_name_full (target_class, member_name, NULL);
+	if (target_field == NULL || target_field->type->type != ret_type->type)
+		mono_mb_emit_exception_full (mb, "System", "MissingFieldException", "UnsafeAccessor");
+
+	mono_mb_emit_op (mb, CEE_LDFLDA, target_field);
+	mono_mb_emit_byte (mb, CEE_RET);
+}
+
+static void
+emit_unsafe_accessor_wrapper_ilgen (MonoMethodBuilder *mb, MonoMethod *accessor_method, MonoMethodSignature *sig, MonoGenericContext *ctx, MonoUnsafeAccessorKind kind, const char *member_name)
+{
+	switch (kind) {
+	case MONO_UNSAFE_ACCESSOR_FIELD:
+	case MONO_UNSAFE_ACCESSOR_STATIC_FIELD:
+		emit_unsafe_accessor_field_wrapper (mb, accessor_method, sig, ctx, kind, member_name);
+		return;
+	case MONO_UNSAFE_ACCESSOR_CTOR:
+		// TODO
+		mono_mb_emit_byte (mb, CEE_RET);
+		return;
+	case MONO_UNSAFE_ACCESSOR_METHOD:
+	case MONO_UNSAFE_ACCESSOR_STATIC_METHOD:
+		// TODO
+		mono_mb_emit_byte (mb, CEE_RET);
```

Comment:
Maybe 

```c
mono_mb_emit_exception_full (mb, "System", "NotImplementedException", "UnsafeAccessor");
```

for now so that it is really obvious if someone tries to use it